### ISSUE TITLE
Error in rule AberrantSplicing_pipeline_Counting_01_1_countRNA_splitReads_samplewise_R

### ISSUE THREAD

**Issue by tomh1lll** [gagneurlab/drop#210]
I am able to run DROP on the demo dataset, but once I change the data to real data I get the following error in the aberrantSplicing pipeline. This error does not appear during the dry run & does not seem to be related to the data. Is there an easy to identify way to fix this problem?

```
Error in rule AberrantSplicing_pipeline_Counting_01_1_countRNA_splitReads_samplewise_R:
    jobid: 48
    output: /gpfs/gsfs11/users/NCBR/projects/NCBR-171/demo/Output/processed_data/aberrant_splicing/datasets/cache/raw-fraser/sample_tmp/splitCounts/sample_WB_S13_S13.done
    log: /gpfs/gsfs11/users/NCBR/projects/NCBR-171/demo/.drop/tmp/AS/fraser/splitReads/WB_S13_S13.Rds (check log file(s) for error message)

RuleException:
CalledProcessError in line 128 of /tmp/tmpvrh3irwr:
Command 'set -euo pipefail;  Rscript --vanilla /gpfs/gsfs11/users/NCBR/projects/NCBR-171/demo/.snakemake/scripts/tmpj0175l7b.01_1_countRNA_splitReads_samplewise.R' returned non-zero exit status 1.
  File "/usr/local/Anaconda/envs/py3.7/lib/python3.7/site-packages/snakemake/executors/__init__.py", line 2338, in run_wrapper
  File "/tmp/tmpvrh3irwr", line 128, in __rule_AberrantSplicing_pipeline_Counting_01_1_countRNA_splitReads_samplewise_R
  File "/usr/local/Anaconda/envs/py3.7/lib/python3.7/site-packages/snakemake/executors/__init__.py", line 566, in _callback
  File "/usr/local/Anaconda/envs/py3.7/lib/python3.7/concurrent/futures/thread.py", line 57, in run
  File "/usr/local/Anaconda/envs/py3.7/lib/python3.7/site-packages/snakemake/executors/__init__.py", line 552, in cached_or_run
  File "/usr/local/Anaconda/envs/py3.7/lib/python3.7/site-packages/snakemake/executors/__init__.py", line 2350, in run_wrapper
Warning message:
In countSplitReads(sampleID = sample_id, fds = fds, NcpuPerSample = snakemake@threads,  :
  Not counting chromosome(s) chrMT in sample WB_S33_S33 as it is not specified in the provided genome.
Wed Jun 16 15:26:05 2021: fraser, WB_S33_S33 no. splice junctions (split counts) = 251206
```

I also see this error in the aberrantExpression pipeline:

```
Error in rule AberrantExpression_pipeline_Counting_countReads_R:
    jobid: 41
    output: /gpfs/gsfs11/users/NCBR/projects/NCBR-171/demo/Output/processed_data/aberrant_expression/v30/counts/WB_S25_S25.Rds
    log: /gpfs/gsfs11/users/NCBR/projects/NCBR-171/demo/.drop/tmp/AE/v30/counts/WB_S25_S25.Rds (check log file(s) for error message)

RuleException:
CalledProcessError in line 34 of /tmp/tmp40g312vt:
Command 'set -euo pipefail;  Rscript --vanilla /gpfs/gsfs11/users/NCBR/projects/NCBR-171/demo/.snakemake/scripts/tmp115arnxx.countReads.R' returned non-zero exit status 137.
  File "/usr/local/Anaconda/envs/py3.7/lib/python3.7/site-packages/snakemake/executors/__init__.py", line 2338, in run_wrapper
  File "/tmp/tmp40g312vt", line 34, in __rule_AberrantExpression_pipeline_Counting_countReads_R
  File "/usr/local/Anaconda/envs/py3.7/lib/python3.7/site-packages/snakemake/executors/__init__.py", line 566, in _callback
  File "/usr/local/Anaconda/envs/py3.7/lib/python3.7/concurrent/futures/thread.py", line 57, in run
  File "/usr/local/Anaconda/envs/py3.7/lib/python3.7/site-packages/snakemake/executors/__init__.py", line 552, in cached_or_run
  File "/usr/local/Anaconda/envs/py3.7/lib/python3.7/site-packages/snakemake/executors/__init__.py", line 2350, in run_wrapper
```

**Comment by vyepez88** [comment]
Hi, these errors where it simply say that the script didn't work and returned non-zero exit status are usually caused by lack of memory. Be sure that your server has enough memory and not to send too many samples in parallel. Each core counts one sample in the aberrant expression module, and 3 cores count one sample in the aberrant splicing module.

**Comment by tomh1lll** [comment]
> Hi, these errors where it simply say that the script didn't work and returned non-zero exit status are usually caused by lack of memory. Be sure that your server has enough memory and not to send too many samples in parallel. Each core counts one sample in the aberrant expression module, and 3 cores count one sample in the aberrant splicing module.

Thank you for the help, that seems to have fixed this error, however I now get another error in the FRASER steps of aberrant splicing that I'm not sure what to make of:

```
Error in vecseq(f__, len__, if (allow.cartesian || notjoin || !anyDuplicated(f__,  : 
  Join results in 17223490 rows; more than 1350593 = nrow(x)+nrow(i). Check for duplicate key values in i each of which join to the same group in x over and over again. If that's ok, try by=.EACHI to run j for each group to avoid the large allocation. If you are sure you wish to proceed, rerun with allow.cartesian=TRUE. Otherwise, please search for this error message in the FAQ, Wiki, Stack Overflow and data.table issue tracker for advice.
Calls: add_HPO_cols ... merge -> merge.data.table -> [ -> [.data.table -> vecseq
Execution halted
[Mon Jun 21 13:54:07 2021]
Error in rule AberrantSplicing_pipeline_FRASER_07_extract_results_FraseR_R:
    jobid: 0
    output: /gpfs/gsfs11/users/NCBR/projects/NCBR-171/demo/Output/processed_results/aberrant_splicing/results/v38/fraser/fraser/results_per_junction.tsv, /gpfs/gsfs11/users/NCBR/projects/NCBR-171/demo/Output/processed_results/aberrant_splicing/results/v38/fraser/fraser/results.tsv, /gpfs/gsfs11/users/NCBR/projects/NCBR-171/demo/Output/processed_results/aberrant_splicing/datasets/savedObjects/fraser--v38/fds-object.RDS
    log: /gpfs/gsfs11/users/NCBR/projects/NCBR-171/demo/.drop/tmp/AS/fraser--v38/07_results.Rds (check log file(s) for error message)

RuleException:
CalledProcessError in line 240 of /tmp/tmpue7o8_aw:
Command 'set -euo pipefail;  Rscript --vanilla /gpfs/gsfs11/users/NCBR/projects/NCBR-171/demo/.snakemake/scripts/tmp9gw8ku4_.07_extract_results_FraseR.R' returned non-zero exit status 1.
  File "/usr/local/apps/snakemake/conda/envs/5.24.1/lib/python3.7/site-packages/snakemake/executors/__init__.py", line 2189, in run_wrapper
  File "/tmp/tmpue7o8_aw", line 240, in __rule_AberrantSplicing_pipeline_FRASER_07_extract_results_FraseR_R
  File "/usr/local/apps/snakemake/conda/envs/5.24.1/lib/python3.7/site-packages/snakemake/executors/__init__.py", line 529, in _callback
  File "/usr/local/apps/snakemake/conda/envs/5.24.1/lib/python3.7/concurrent/futures/thread.py", line 57, in run
  File "/usr/local/apps/snakemake/conda/envs/5.24.1/lib/python3.7/site-packages/snakemake/executors/__init__.py", line 515, in cached_or_run
  File "/usr/local/apps/snakemake/conda/envs/5.24.1/lib/python3.7/site-packages/snakemake/executors/__init__.py", line 2201, in run_wrapper
```

If you could impart any information on how to fix this it would be much appreciated. The error appears to be the same as here: https://github.com/gagneurlab/drop/issues/213

**Comment by nickhsmith** [comment]
I'm glad you've made some progress. I addressed issue #213  and would appreciate if you sent the following as well:

"It seems like there is an issue with the adding of the HPO columns from the sample annotation table to match the resulting genes with the potential phenotypic labels. Could you please send us your sample annotation table, I'm guessing the problem is that something has gone wrong in the HPO column. "

**Comment by tomh1lll** [comment]
Thank you for the quick reply and for helping with this issue.

I currently haven't used the HPO column properly and so have used one of the annotations from the test data (sample file attached as a text file).
[sample_annotation_test.txt](https://github.com/gagneurlab/drop/files/6688907/sample_annotation_test.txt)
I can find the correct HPO for samples or remove HPOs if those are preferable.

**Comment by nickhsmith** [comment]
If you remove the column it should work without this problem. If the HPO_terms are just fillers (and don't represent the HPO terms associated with the sample) just delete that column, or all of the values in it.

**Comment by nickhsmith** [comment]
Hopefully, this solved your issue. If not please let us know or start a new issue. If I don't hear back I will close the issue towards the end of the week

**Comment by tomh1lll** [comment]
Thank you for this suggestion, this seems to have fixed the error and the aberrantSplicing pipeline is currently working as intended.
However, I am still finding an error with the aberrantExpression pipeline in the step 

```
Error in rule AberrantExpression_pipeline_Counting_countReads_R:
    jobid: 0
    output: /gpfs/gsfs11/users/NCBR/projects/NCBR-171/demo3/Output/processed_data/aberrant_expression/v38/counts/WB_S1_S1.Rds
    log: /gpfs/gsfs11/users/NCBR/projects/NCBR-171/demo3/.drop/tmp/AE/v38/counts/WB_S1_S1.Rds (check log file(s) for error message)

RuleException:
CalledProcessError in line 34 of /tmp/tmpa0wxwjip:
Command 'set -euo pipefail;  Rscript --vanilla /gpfs/gsfs11/users/NCBR/projects/NCBR-171/demo3/.snakemake/scripts/tmp97f1qv1u.countReads.R' returned non-zero exit status 137.
  File "/usr/local/apps/snakemake/conda/envs/5.24.1/lib/python3.7/site-packages/snakemake/executors/__init__.py", line 2189, in run_wrapper
  File "/tmp/tmpa0wxwjip", line 34, in __rule_AberrantExpression_pipeline_Counting_countReads_R
  File "/usr/local/apps/snakemake/conda/envs/5.24.1/lib/python3.7/site-packages/snakemake/executors/__init__.py", line 529, in _callback
  File "/usr/local/apps/snakemake/conda/envs/5.24.1/lib/python3.7/concurrent/futures/thread.py", line 57, in run
  File "/usr/local/apps/snakemake/conda/envs/5.24.1/lib/python3.7/site-packages/snakemake/executors/__init__.py", line 515, in cached_or_run
  File "/usr/local/apps/snakemake/conda/envs/5.24.1/lib/python3.7/site-packages/snakemake/executors/__init__.py", line 2201, in run_wrapper
Shutting down, this might take some time.
Exiting because a job execution failed. Look above for error message
```
